lib/aws/request.js

```diff
@@ -30,7 +31,11 @@ const proxy =
 
 const proxyOptions = {};
 if (proxy) {
-  Object.assign(proxyOptions, new URL(proxy));
+  // TODO: switch to recomended WHATWG URL
```

Comment:
Let's remove "TODO" mark (our current convention is to put such comments only when we have coined when exactly given change should be applied - and in most cases it's when something waits for a new major, then when preparing new major branch, we search for all "TODO" and address them).

This comment should be more like:

_Not relying on recommended WHATHWG URL, due to missing support for it in..._